packages/grafana-ui/src/components/uPlot/types.ts

```diff
@@ -30,7 +30,13 @@ export abstract class PlotConfigBuilder<P, T> {
   abstract getConfig(): T;
 }
 
+export interface FieldIndexRef {
+  frameIndex: number;
+  fieldIndex: number;
+}
+
 export interface AlignedFrameWithGapTest {
   frame: DataFrame;
   isGap: Series.isGap;
+  getFieldIndexRef: (alignedFieldIndex: number) => FieldIndexRef | undefined;
```

Comment:
Wonder if we shouldn't throw for not existent indexes rather than returning `undefined`? I guess getFieldIndex range will always return something for a valid index and accessing anything out of range could be potentially just a bug in the code.